code/controllers/subsystem/addictions.dm

```diff
@@ -0,0 +1,19 @@
+/*!
+This subsystem mostly exists to populate and manage the withdrawal singletons.
+*/
+
+SUBSYSTEM_DEF(addiction)
+	name = "Addiction"
+	flags = SS_NO_FIRE
+	///Dictionary of addiction.type || addiction ref
+	var/list/all_addictions = list()
+
+/datum/controller/subsystem/addiction/Initialize(timeofday)
+	InitializeAddictions()
+	return ..()
+
+///Ran on initialize, populates the addiction dictionary
+/datum/controller/subsystem/addiction/proc/InitializeAddictions()
+	for(var/type in subtypesof(/datum/addiction))
+		var/datum/addiction/ref = new type
+		all_addictions[type] = ref
```

Comment:
The fuck is this useless subsystem, why isn't this just a global list or a lazy init list

Reply:
yanno im just gonna trust tgcoders on this. subsystems are technically more performant for init